### ISSUE TITLE
Allow for environment specific encodings

### ISSUE THREAD

**Issue by brizzbuzz** [Just-Insane/helm-vault#25]
## Is your feature request related to a problem? Please describe.

Hey, apologies if this is already something that has been brought up, a quick search didn't show anything.  Not a problem, just a plain ol' feature request 

## Describe the solution you'd like

I think it would be _super_ useful to be able to specify the environment that you would like to encode for.  IE I have a `values.yaml` file that includes a `database_password` property.  It seems reasonable that I would want this password to be different for each environment (sandbox, staging, production, etc.).  Perhaps then decoded as `values.yaml.production.dec` for example.

## Describe alternatives you've considered

I think the most immediate solution would be separate `value.yaml`, `value.sandbox.yaml` and so on, but this isn't very DRY and could lead to issues across environments.  

## Additional context

Nothing else, just think this would be cool 😃

**Comment by brizzbuzz** [comment]
Ok went an educated myself a bit on how this works... I'm using Vault KV V2, not sure if V1 differs on this front.  It seems that for a secret `image.repository` in a repo called `helm-starter-go`, said secret would be stored in Vault as follows

```
❯ vault kv get secret/secret/helm/helm-starter-go/image/repository
====== Metadata ======
Key              Value
---              -----
created_time     2021-03-16T16:40:39.445973Z
deletion_time    n/a
destroyed        false
version          1

==== Data ====
Key      Value
---      -----
value    testerino
```

Perhaps a logic place for this to be indicated in the path would be right after the repo declaration? 

```
secret/secret/helm/helm-starter-go/{environment}/image/repository
```

**Comment by Just-Insane** [comment]
I like this idea.

It should be easy enough to add a flag to each of the commands which appends the provided value to the path.

I haven't done much on this project lately, but if I find some time I'll try and get through all of these open tickets.

**Comment by brizzbuzz** [comment]
Awesome, I can also try to find some time to tackle this, it seems like a really good intro ticket, though my python skills are... rusty bordering non-existent 😅

**Comment by Just-Insane** [comment]
Fixed by #27
